go/service/systemd_linux.go

```diff
@@ -5,11 +5,25 @@ package service
 import (
 	"errors"
 	"net"
+	"os"
 
 	sdActivation "github.com/coreos/go-systemd/activation"
 	sdDaemon "github.com/coreos/go-systemd/daemon"
 )
 
+var isSocketActivated bool
+
+func init() {
+	// We read this env var during startup, because
```

Comment:
Is this the best place to do this?  I'm always suspicious of items running in `init()`